docs/ai/how-to/snippets/semantic-kernel/LocalModelExamples.cs

```diff
@@ -0,0 +1,127 @@
+        var service = new MyChatCompletionService();
+        builder.Services.AddKeyedSingleton<IChatCompletionService>("myChatService1", service);
+
+        // Alternatively, add your chat completion service as a factory method
+        builder.Services.AddKeyedSingleton<IChatCompletionService>("myChatService2", (_, _) => new MyChatCompletionService());
+
+        // Add any other Kernel services or configurations
+        // ...
+        Kernel kernel = builder.Build();
+        // </addChatService>
+
+        return kernel;
+    }
+
+    static async Task UseTextGenerationServiceExample()
+    {
+        IKernelBuilder builder = Kernel.CreateBuilder();
+        builder.Services.AddKeyedSingleton<ITextGenerationService>("myTextService", new MyTextGenerationService());
+        Kernel kernel = builder.Build();
+
+        // <useTextService>
+        var executionSettings = new PromptExecutionSettings
+        {
+            // Add execution settings, such as the ModelID and ExtensionData
+            ModelId = "MyModelId",
+            ExtensionData = new Dictionary<string, object> { { "MaxTokens", 500 } }
+        };
+
+        // Send a prompt to your model directly through the Kernel
+        string prompt = "Please list three services offered by Azure";
+        string? response = await kernel.InvokePromptAsync<string>(prompt);
+        Console.WriteLine($"Output: {response}");
+
+        // Alteratively, send a prompt to your model through the text generation service
+        ITextGenerationService textService = kernel.GetRequiredService<ITextGenerationService>();
+        TextContent responseContents = await textService.GetTextContentAsync(prompt, executionSettings);
+        Console.WriteLine($"Output: {responseContents.Text}");
+        // </useTextService>
+    }
+
+    static async Task UseChatCompletionServiceExample()
+    {
+        IKernelBuilder builder = Kernel.CreateBuilder();
+        builder.Services.AddKeyedSingleton<IChatCompletionService>("myChatService", new MyChatCompletionService());
+        Kernel kernel = builder.Build();
+
+        // <useChatService>
+        var executionSettings = new PromptExecutionSettings
+        {
+            // Add execution settings, such as the ModelID and ExtensionData
+            ModelId = "MyModelId",
+            ExtensionData = new Dictionary<string, object> { { "MaxTokens", 500 } }
+        };
+
+        // Send a string representation of the chat history to your model directly through the Kernel
+        // This uses a special syntax to denote the role for each message
+        // For more information on this syntax see https://learn.microsoft.com/en-us/semantic-kernel/prompts/your-first-prompt?tabs=Csharp#6-using-message-roles-in-chat-completion-prompts
+        string prompt = @"
+        <message role=""system"">the initial system message for your chat history</message>
+        <message role=""user"">the user's initial message</message>";
```

Comment:
Yeah, this gets a lot easier with raw string literals:

```suggestion
        string prompt = """
        <message role="system">the initial system message for your chat history</message>
        <message role="user">the user's initial message</message>
        """;
```